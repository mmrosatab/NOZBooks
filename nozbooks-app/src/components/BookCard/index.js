import React from "react";
import Container from "./styles";
import imagePlaceholder from "../../assets/images/image_placeholder.png";
import { Card, CardContent, CardActionArea } from "@mui/material";
import { handIcon } from "../../constants/images";

function BookCard({ book, handleClickModal }) {
  const image = book.imageUrl == null ? imagePlaceholder : book.imageUrl;
  const { title, authors, pageCount, publisher, published } = book;

  return (
    <Card key={book.id}>
      <CardActionArea
        style={styleCardContent}
        onClick={() => handleClickModal(book.id)}
      >
        <CardContent>
          <Container>
            <div className="image-book">
              <img src={image} alt="book" />
            </div>
            <div className="infos">
              <div className="info-title-authors">
                <div>
                  <p className="p-title">{title}</p>
                </div>
                <div>
                  {authors.length > 1 ? (
                    authors.map((author, index) => {
                      return (
                        <p className="p-authors" key={index}>
                          {author}
                        </p>
                      );
                    })
                  ) : (
                    <p className="p-author">{authors[0]}</p>
                  )}
                </div>
              </div>

              <div className="info-book">
                <div>
                  <p className="p-page-count">{`${pageCount} p√°ginas`}</p>
                </div>
                <div>
                  <p className="p-publisher">{publisher}</p>
                </div>
                <div>
                  <p className="p-published">{`Publicado em ${published} `}</p>
                </div>
              </div>
            </div>
          </Container>
        </CardContent>
      </CardActionArea>
    </Card>
  );
}

export default BookCard;

const styleCardContent = {
  cursor: `url(${handIcon}), auto`,
};
