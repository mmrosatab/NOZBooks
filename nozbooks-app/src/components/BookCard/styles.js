import styled from "styled-components";

const Container = styled.div`
  display: flex;
  height: 100px;

  .image-book img {
    max-height: 100%;
  }

  .infos {
    margin-left: 8%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }

  .info-title-authors {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
  }

  .info-book {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
  }

  .p-title {
    font-weight: 500;
    line-height: 20px;
    color: #333333;
  }

  .p-authors,
  .p-author {
    color: #2e63f7;
    font-weight: 400;
  }

  .p-page-count,
  .p-publisher,
  .p-published {
    color: #999999;
  }

  @media screen and (min-width: 768px) {
    .p-title {
      font-size: calc(0.1rem + 0.7vw);
    }

    .p-authors,
    .p-author {
      font-size: calc(0.1rem + 0.6vw);
    }

    .p-page-count,
    .p-publisher,
    .p-published {
      font-size: calc(0.1rem + 0.6vw);
    }
  }

  @media screen and (max-width: 767px) {
    .p-title {
      font-size: calc(0.86rem);
    }

    .p-authors,
    .p-author {
      font-size: calc(0.75rem);
    }

    .p-page-count,
    .p-publisher,
    .p-published {
      font-size: calc(0.85rem);
    }
  }
`;

export default Container;
