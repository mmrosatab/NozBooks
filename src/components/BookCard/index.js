import React from "react";
import Container from "./styles";
import imagePlaceholder from "../../assets/images/image_placeholder.png";
import { Card, CardContent, CardActionArea } from "@mui/material";
import { handIcon } from "../../constants/images";

function BookCard({ book, handleClickModal }) {
  const image = book.imageUrl === null ? imagePlaceholder : book.imageUrl;
  const { title, authors, pageCount, publisher, published } = book;

  return (
    <Card data-testid="card-testid" key={book.id} sx={{ height: "100%" }}>
      <CardActionArea
        data-testid="card-area-testid"
        style={styleCardContent}
        onClick={() => handleClickModal(book.id)}
      >
        <CardContent sx={{ height: "100%" }}>
          <Container>
            <div className="image-book">
              <img src={image} alt="book" />
            </div>
            <div className="infos">
              <div className="info-title-authors">
                <div>
                  <p className="p-title">{title}</p>
                </div>
                <div>
                  {authors.length > 1 ? (
                    authors.map((author, index) => {
                      return (
                        <p className="p-authors" key={index}>
                          {author}
                        </p>
                      );
                    })
                  ) : (
                    <p className="p-author">{authors[0]}</p>
                  )}
                </div>
              </div>

              <div className="info-book">
                <div>
                  <p className="p-page-count">{`${pageCount} p√°ginas`}</p>
                </div>
                <div>
                  <p className="p-publisher">{publisher}</p>
                </div>
                <div>
                  <p className="p-published">{`Publicado em ${published} `}</p>
                </div>
              </div>
            </div>
          </Container>
        </CardContent>
      </CardActionArea>
    </Card>
  );
}

export default BookCard;

const styleCardContent = {
  cursor: `url(${handIcon}), auto`,
  height: "100%",
};
