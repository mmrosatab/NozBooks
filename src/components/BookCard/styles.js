import styled from "styled-components";

const Container = styled.div`
  display: flex;
  height: 100%;
  width: 100%;

  .image-book img {
    max-height: 100%;
  }
  .infos {
    margin-left: 8%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
  .info-title-authors {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
  }
  .info-book {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
  }
  .p-title {
    font-weight: 500;
    line-height: 20px;
    color: #333333;
  }
  .p-authors,
  .p-author {
    color: #2e63f7;
    font-weight: 400;
  }
  .p-page-count,
  .p-publisher,
  .p-published {
    color: #999999;
  }

  /*  Tablets */
  @media screen and (min-width: 768px) and (max-width: 899px) {
    .p-title {
      font-size: calc(0.1rem + 1vw);
    }
    .p-authors,
    .p-author {
      font-size: calc(0.1rem + 0.9vw);
    }
    .p-page-count,
    .p-publisher,
    .p-published {
      font-size: calc(0.1rem + 0.9vw);
    }
  }

  /* Desktop Full Size */
  @media screen and (min-width: 900px) {
    .p-title {
      font-size: calc(0.1rem + 0.7vw);
    }
    .p-authors,
    .p-author {
      font-size: calc(0.1rem + 0.6vw);
    }
    .p-page-count,
    .p-publisher,
    .p-published {
      font-size: calc(0.1rem + 0.6vw);
    }
  }

  /* Mobile */
  @media screen and (max-width: 767px) {
    .p-title {
      font-size: calc(0.86rem);
    }
    .p-authors,
    .p-author {
      font-size: calc(0.75rem);
    }
    .p-page-count,
    .p-publisher,
    .p-published {
      font-size: calc(0.7rem);
    }
  }
`;

export default Container;
