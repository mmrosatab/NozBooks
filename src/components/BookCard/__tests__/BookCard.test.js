import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import BookCard from "..";

const bookMock = {
  authors: ["Júlio César Carvalho", "Márcia Xavier", "Ígor Carvalho"],
  title: "Aliquam et",
  description:
    "Aut beatae deserunt omnis maiores tempora possimus. Consequuntur cumque ut reiciendis dolor. Tempore tempore consequatur id eum cumque est est facere totam. Unde est sint ut repudiandae officia est odit. Repellat non rerum suscipit. Aspernatur et ut quidem et eos voluptatem aliquam.\n \rSuscipit exercitationem officiis voluptatem fuga blanditiis laboriosam nihil. Ipsa magnam in iste ut est possimus perspiciatis mollitia fugiat. Nemo qui officia quisquam sed facilis quia. Quia iure sit quas dolorem nam sit neque.",
  pageCount: 947,
  category: "Biografias",
  imageUrl: "https://d2drtqy2ezsot0.cloudfront.net/appnoz/Book-8.jpg",
  language: "Inglês",
  isbn10: "7143569875",
  isbn13: "599-7143569875",
  publisher: "Albuquerque S.A.",
  published: 2021,
  id: "61c9c5246189745dcd504e2d",
};

const handleClickModal = jest.fn();

const renderComponent = () => {
  return render(<BookCard book={bookMock} handleClickModal={handleClickModal} />);
};

describe("BookCard Tests", () => {
  test("Should render correctly", () => {
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  test("Check BookCard is visible", () => {
    renderComponent();
    const bookCard = screen.getByTestId("card-testid");
    expect(bookCard).toBeVisible();
    expect(bookCard).toBeInTheDocument();
  });

  test("Check handleClickModal is called after click on modal", () => {
    renderComponent();
    const bookCard = screen.getByTestId("card-area-testid");
    fireEvent.click(bookCard);
    expect(handleClickModal).toHaveBeenCalledTimes(1);
  });
});
