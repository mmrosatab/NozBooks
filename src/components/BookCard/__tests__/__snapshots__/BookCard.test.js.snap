// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard Tests Should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
    data-testid="card-testid"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
      data-testid="card-area-testid"
      style="cursor: url(hand.png), auto; height: 100%;"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root css-16d4f6n-MuiCardContent-root"
      >
        <div
          class="sc-bdvvtL bukMTk"
        >
          <div
            class="image-book"
          >
            <img
              alt="book"
              src="https://d2drtqy2ezsot0.cloudfront.net/appnoz/Book-8.jpg"
            />
          </div>
          <div
            class="infos"
          >
            <div
              class="info-title-authors"
            >
              <div>
                <p
                  class="p-title"
                >
                  Aliquam et
                </p>
              </div>
              <div>
                <p
                  class="p-authors"
                >
                  Júlio César Carvalho
                </p>
                <p
                  class="p-authors"
                >
                  Márcia Xavier
                </p>
                <p
                  class="p-authors"
                >
                  Ígor Carvalho
                </p>
              </div>
            </div>
            <div
              class="info-book"
            >
              <div>
                <p
                  class="p-page-count"
                >
                  947 páginas
                </p>
              </div>
              <div>
                <p
                  class="p-publisher"
                >
                  Albuquerque S.A.
                </p>
              </div>
              <div>
                <p
                  class="p-published"
                >
                  Publicado em 2021 
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
